x/gamm/keeper/pool_service_test.go

```diff
@@ -222,6 +222,121 @@ func (suite *KeeperTestSuite) TestCreateBalancerPool() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestInitializePool() {
+	params := suite.App.GAMMKeeper.GetParams(suite.Ctx)
+	testAccount := suite.TestAccs[0]
+
+	// get raw pool creation fee(s) as DecCoins
+	poolCreationFeeDecCoins := sdk.DecCoins{}
+	for _, coin := range params.PoolCreationFee {
+		poolCreationFeeDecCoins = poolCreationFeeDecCoins.Add(sdk.NewDecCoin(coin.Denom, coin.Amount))
+	}
+
+	tests := []struct {
+		name        string
+		msg         balancertypes.MsgCreateBalancerPool
+		emptySender bool
+		expectPass  bool
+	}{
+		{
+			name:        "initialize pool with default assets",
+			msg:         balancer.NewMsgCreateBalancerPool(testAccount, defaultPoolParams, defaultPoolAssets, defaultFutureGovernor),
+			emptySender: false,
+			expectPass:  true,
+		},
+		{
+			name:        "initialize pool with empty sender",
+			msg:         balancer.NewMsgCreateBalancerPool(testAccount, defaultPoolParams, defaultPoolAssets, defaultFutureGovernor),
+			emptySender: true,
+			expectPass:  false,
```

Comment:
Can we please also have the following tests:
- stableswap pool - gets initialized
- any non-CFMM pool such as concentrated liquidity does not get initialized. You can find mocks in `tests/mocks` helpful for this

Reply:
sure. Lemme check it

Reply:
I don't really sure why you want a CL pool initialized here in a test of ``x/gamm``. I think it is better placed in ``x/concentrated-liquidity``. Do you agree ?

Reply:
I think we want to test that if you give it a CL pool (or any non-CFMM pool / mock), then it is going to fail. This function should only execute without error if a CFMM pool is given to it. The pool that implements the following interface:
https://github.com/osmosis-labs/osmosis/blob/f2a497dfac7f41878ddbdcb2d1df4acbe8e2e9b6/x/gamm/types/pool.go#L15

Reply:
Yep. If I use the function gamm.InitializePool for a mock CL pool. It will not fail.

Reply:
Did you use `MockConcentratedPoolExtension`?

Reply:
Yes. I did. You can see in the files changes

Reply:
The correct behavior of the CL pool test case is to panic. I have updated the code